packages/frontend-shared/src/gql-components/topnav/LoginModal.cy.tsx

```diff
@@ -36,44 +36,103 @@ const mountSuccess = (viewer: TestCloudViewer = cloudViewer) => {
+      cy.findByRole('button', { name: text.login.actionOpening })
+      .should('be.visible')
+      .and('be.disabled')
     })
 
-    cy.contains('h2', text.login.titleInitial).should('be.visible')
+    it('shows correct "waiting for login" status', () => {
+      cy.mountFragment(LoginModalFragmentDoc, {
+        render: (gqlVal) => {
+          gqlVal.authState.browserOpened = true
+
+          return <div class="border-current border-1 h-700px resize overflow-auto"><LoginModal gql={gqlVal} modelValue={true} /></div>
+        },
+      })
+
+      cy.findByRole('button', { name: text.login.actionLogin }).click()
 
-    // begin the login process
-    cy.findByRole('button', { name: text.login.actionLogin }).click()
+      cy.findByRole('button', { name: text.login.actionWaiting })
+      .should('be.visible')
+      .and('be.disabled')
+    })
 
-    // ensure we reach "browser is opening" status on the CTA
-    cy.findByRole('button', { name: text.login.actionOpening })
-    .should('be.visible')
-    .and('be.disabled')
+    it('shows successful login status', () => {
+      mountSuccess()
+      cy.contains('h2', text.login.titleSuccess).should('be.visible')
+      cy.contains(text.login.bodySuccess.replace('{0}', cloudViewer.fullName)).should('be.visible')
+      cy.contains('a', cloudViewer.fullName).should('have.attr', 'href', 'https://on.cypress.io/dashboard/profile')
+    })
   })
 
-  it('shows correct "waiting for login" status', () => {
-    cy.mountFragment(LoginModalFragmentDoc, {
-      onResult: (result, ctx) => {
-        ctx.isAuthBrowserOpened = true
-        result.__typename = 'Query'
-        result.isAuthBrowserOpened = true
-      },
-      render: (gqlVal) => <div class="border-current border-1 h-700px resize overflow-auto"><LoginModal gql={gqlVal} modelValue={true} /></div>,
+  describe('errors', () => {
+    it('shows an error state when browser cannot be launched', () => {
+      const authUrl = 'http://127.0.0.1:0000/redirect-to-auth'
+
+      cy.mountFragment(LoginModalFragmentDoc, {
+        onResult: (result) => {
+          result.__typename = 'Query'
+          result.authState.name = 'AUTH_COULD_NOT_LAUNCH_BROWSER'
+          result.authState.message = 'http://127.0.0.1:0000/redirect-to-auth'
+        },
+        render: (gqlVal) =>
+          (<div class="border-current border-1 h-700px resize overflow-auto">
+            <LoginModal gql={gqlVal} modelValue={true}/>
+          </div>),
+      })
+
+      cy.contains('button', text.login.actionTryAgain).should('not.exist')
+      cy.contains('button', text.login.actionCancel).should('not.exist')
+      cy.contains(text.login.titleBrowserError).should('be.visible')
+      cy.contains(text.login.bodyBrowserError).should('be.visible')
+      cy.contains(text.login.bodyBrowserErrorDetails).should('be.visible')
+      cy.contains(authUrl).should('be.visible')
+      cy.contains('button', 'Copy').should('be.visible')
```

Comment:
I see your point. How I about I add "skip if Firefox" to https://cypress-io.atlassian.net/browse/UNIFY-999 which deals with a different problematic aspects of testing the copy functionality. We'd probably end up with something like our `validatedExternalLink` custom command, but for what gets put in the clipboard.

Reply:
I went ahead and added some detail to that ticket. The `cy.contains(authUrl).should('be.visible')` covers the test we want to copy, and since the button is in our requirements for the component, I'd rather keep asserting it until we do hit that FF problem.